File: app/src/main/java/com/hellowo/teamfinder/fcm/MessagingService.java
package com.hellowo.teamfinder.fcm;

import android.app.NotificationManager;
import android.app.PendingIntent;
import android.content.Context;
import android.content.Intent;
import android.media.RingtoneManager;
import android.net.Uri;
import android.support.v4.app.NotificationCompat;
import android.util.Log;

import com.google.firebase.messaging.RemoteMessage;
import com.hellowo.teamfinder.R;
import com.hellowo.teamfinder.ui.activity.MainActivity;

import org.json.JSONException;
import org.json.JSONObject;

import java.util.Map;

public class MessagingService extends com.google.firebase.messaging.FirebaseMessagingService {

    private static final String TAG = "MyFirebaseMsgService";

    @Override
    public void onMessageReceived(RemoteMessage remoteMessage) {
        Log.d(TAG, "From: " + remoteMessage.getFrom());
        if (remoteMessage.getData().size() > 0) {
            try {
                JSONObject data = new JSONObject(remoteMessage.getData().get("data"));
                String subject = data.getString("subject");
                String message = data.getString("message");
                Log.d("aaa", subject + "/" + message);
                sendNotification(subject, message);
            } catch (JSONException e) {
                e.printStackTrace();
            }
        }
    }

    private void sendNotification(String subject, String message) {
        Intent intent = new Intent(this, MainActivity.class);
        intent.addFlags(Intent.FLAG_ACTIVITY_CLEAR_TOP);
        PendingIntent pendingIntent = PendingIntent.getActivity(this, 0, intent,
                PendingIntent.FLAG_ONE_SHOT);

        Uri defaultSoundUri= RingtoneManager.getDefaultUri(RingtoneManager.TYPE_NOTIFICATION);
        NotificationCompat.Builder notificationBuilder = new NotificationCompat.Builder(this)
                .setSmallIcon(R.mipmap.ic_launcher_round)
                .setColor(getColor(R.color.colorPrimary))
                .setContentTitle(subject)
                .setContentText(message)
                .setAutoCancel(true)
                .setSound(defaultSoundUri)
                .setContentIntent(pendingIntent);

        NotificationManager notificationManager =
                (NotificationManager) getSystemService(Context.NOTIFICATION_SERVICE);

        notificationManager.notify(0, notificationBuilder.build());
    }
}

File: app/src/main/java/com/hellowo/teamfinder/fcm/MessagingService.kt
package com.hellowo.teamfinder.fcm

import android.app.NotificationManager
import android.app.PendingIntent
import android.content.Context
import android.content.Intent
import android.graphics.Bitmap
import android.media.RingtoneManager
import android.net.Uri
import android.support.v4.app.NotificationCompat
import android.util.Log
import com.bumptech.glide.Glide
import com.bumptech.glide.request.animation.GlideAnimation
import com.bumptech.glide.request.target.SimpleTarget

import com.google.firebase.messaging.RemoteMessage
import com.hellowo.teamfinder.AppConst
import com.hellowo.teamfinder.R
import com.hellowo.teamfinder.ui.activity.MainActivity
import com.hellowo.teamfinder.utils.PUSH_TYPE_CHAT_MESSAGE
import com.hellowo.teamfinder.utils.makePublicPhotoUrl
import jp.wasabeef.glide.transformations.CropCircleTransformation

import org.json.JSONException
import org.json.JSONObject
import android.content.pm.PackageManager



class MessagingService : com.google.firebase.messaging.FirebaseMessagingService() {

    override fun onMessageReceived(remoteMessage: RemoteMessage?) {
        if (remoteMessage?.data!!.isNotEmpty()) {
            try {
                val data = JSONObject(remoteMessage.data["data"])
                val pushType = data.getInt("pushType")
                when(pushType) {
                    PUSH_TYPE_CHAT_MESSAGE -> makeChatMessageNoti(data)
                }
            } catch (e: JSONException) {
                e.printStackTrace()
            }
        }
    }

    private fun makeChatMessageNoti(data: JSONObject) {
        val userId = data.getString("userId")
        val userName = data.getString("userName")
        val message = data.getString("message")
        val chatId = data.getString("chatId")
        var resource: Bitmap? = null
        try{
            resource = Glide.with(this).load(makePublicPhotoUrl(userId))
                    .asBitmap().transform(CropCircleTransformation(this)).into(100, 100).get()
        }catch (e: Exception){}

        val manager = packageManager
        val intent = manager.getLaunchIntentForPackage(packageName)
        intent.addCategory(Intent.CATEGORY_LAUNCHER)
        intent.putExtra(AppConst.EXTRA_CHAT_ID, chatId)
        val pendingIntent = PendingIntent.getActivity(this,0, intent, PendingIntent.FLAG_ONE_SHOT)

        sendNotification(userName, message, resource, pendingIntent)
    }

    private fun sendNotification(subject: String, message: String, icon: Bitmap?, pIntent: PendingIntent) {
        val defaultSoundUri = RingtoneManager.getDefaultUri(RingtoneManager.TYPE_NOTIFICATION)
        val notificationBuilder = NotificationCompat.Builder(this)
                .setSmallIcon(R.drawable.small_logo)
                .setColor(getColor(R.color.colorPrimary))
                .setContentTitle(subject)
                .setContentText(message)
                .setAutoCancel(true)
                .setSound(defaultSoundUri)
                .setContentIntent(pIntent)

        icon?.let { notificationBuilder.setLargeIcon(it) }

        val notificationManager = getSystemService(Context.NOTIFICATION_SERVICE) as NotificationManager

        notificationManager.notify(0, notificationBuilder.build())
    }
}